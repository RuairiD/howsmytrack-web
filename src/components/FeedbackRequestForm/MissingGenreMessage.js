import React from "react";

const MissingGenreMessage = ({ isRequestTrackless }) => (
    <React.Fragment>
        {isRequestTrackless && "Please select the genre you would like to provide feedback for."}
        {!isRequestTrackless && "Please select a genre for this track. If you are unsure or do not see your genre listed, just select 'No Genre'."}
    </React.Fragment>
);

export default MissingGenreMessage;
