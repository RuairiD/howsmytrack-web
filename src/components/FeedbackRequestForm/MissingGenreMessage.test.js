import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import MissingGenreMessage from "./MissingGenreMessage";

describe("MissingGenreMessage", () => {
    it("renders an error message for a trackless request", () => {
        const wrapper = shallow(
            <MissingGenreMessage
                isRequestTrackless
            />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("renders an error message for a request with a track", () => {
        const wrapper = shallow(
            <MissingGenreMessage />,
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
