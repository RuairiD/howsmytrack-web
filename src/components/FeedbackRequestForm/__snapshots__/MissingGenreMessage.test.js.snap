// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissingGenreMessage renders an error message for a request with a track 1`] = `
<Fragment>
  Please select a genre for this track. If you are unsure or do not see your genre listed, just select 'No Genre'.
</Fragment>
`;

exports[`MissingGenreMessage renders an error message for a trackless request 1`] = `
<Fragment>
  Please select the genre you would like to provide feedback for.
</Fragment>
`;
